clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -670,45 +652,41 @@ class DiagDeviceFunction : public RecursiveASTVisitor<DiagDeviceFunction> {
 
   // Skip checking rules on variables initialized during constant evaluation.
   bool TraverseVarDecl(VarDecl *VD) {
-    ConstexprDepthRAII R(*this, VD->isConstexpr());
+    if (VD->isConstexpr())
```

Comment:
Is this bit necessary?  Does the default behavior not do this right?

Reply:
I will verify this but I think default behavior would traverse VarDecl subgraph if we don't exit early. 

`return true` causes traversal to skip the subgraph and continue with rest of call graph.

Reply:
Ah, yes, I believe you're right.